quic/state/test/AckHandlersTest.cpp

```diff
@@ -1107,6 +1113,74 @@ TEST_P(AckHandlersTest, ImplictAckEventCreation) {
       ackTime);
 }
 
+TEST_P(AckHandlersTest, TestRTTPacketObserverCallback) {
+  QuicServerConnectionState conn(
+      FizzServerQuicHandshakeContext::Builder().build());
+  auto mockCongestionController = std::make_unique<MockCongestionController>();
+  conn.congestionController = std::move(mockCongestionController);
+
+  // Register 1 instrumentation observer
+  auto ib = MockInstrumentationObserver();
+  conn.instrumentationObservers_.emplace_back(&ib);
+
+  PacketNum packetNum = 0;
+  StreamId streamid = 0;
+  TimePoint largestSentTime;
+  while (packetNum < 10) {
+    auto regularPacket = createNewPacket(packetNum, GetParam());
+    WriteStreamFrame frame(streamid++, 0, 0, true);
+    regularPacket.frames.emplace_back(std::move(frame));
+    largestSentTime =
+        Clock::now() - 100ms + std::chrono::milliseconds(packetNum);
+    OutstandingPacket sentPacket(
+        std::move(regularPacket),
+        largestSentTime,
+        1,
+        false /* handshake */,
+        packetNum,
+        packetNum + 1);
+    sentPacket.isAppLimited = false;
+    conn.outstandings.packets.emplace_back(sentPacket);
+    packetNum++;
+  }
+
+  ReadAckFrame ackFrame;
+  ackFrame.largestAcked = 9;
+  ackFrame.ackDelay = 25ms;
+  ackFrame.ackBlocks.emplace_back(0, 9);
+
+  // 0 pending callbacks
+  EXPECT_EQ(0, size(conn.pendingCallbacks));
+
+  auto ackTime = Clock::now() + 25ms;
+  processAckFrame(
+      conn,
+      GetParam(),
+      ackFrame,
+      [](const auto&, const auto&, const auto&) {},
+      [](auto&, auto&, bool) {},
+      ackTime);
+
+  // 1 pending callbacks
+  EXPECT_EQ(1, size(conn.pendingCallbacks));
+
+  auto rttSample = std::chrono::duration_cast<std::chrono::microseconds>(
+      ackTime - largestSentTime);
+  EXPECT_CALL(
+      ib,
+      rttSampleGenerated(AllOf(
+          Field(&InstrumentationObserver::PacketRTT::rttSample, rttSample),
+          Field(&InstrumentationObserver::PacketRTT::ackDelay, 25ms),
+          Field(
+              &InstrumentationObserver::PacketRTT::metadata,
+              Field(&quic::OutstandingPacketMetadata::inflightBytes, 10)))))
```

Comment:
Any way to do a test that involves lossState being updated, so that we can ensure this remains correct?